### PR TITLE
Add spill metrics and improve spill log printing

### PR DESCRIPTION
**Pull request by jacktengg**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [ ] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

File: be/src/vec/spill/spill_writer.cpp [hunk 1]
```diff
@@ -59,6 +59,7 @@ Status SpillWriter::close() {
         COUNTER_UPDATE(_write_file_current_size, meta_.size());
     }
     data_dir_->update_spill_data_usage(meta_.size());
+    ExecEnv::GetInstance()->spill_stream_mgr()->updat_spill_write_bytes(meta_.size());
```
Comment (yiguolei) [inline be/src/vec/spill/spill_writer.cpp:62]: typo error

File: be/src/vec/spill/spill_stream_manager.cpp [hunk 2]
```diff
@@ -84,9 +92,46 @@ Status SpillStreamManager::init() {
...
+            doris::MetricType::COUNTER, doris::MetricUnit::BYTES, "spill_read_bytes");
+    _spill_read_bytes_counter = (IntAtomicCounter*)(_entity->register_metric<IntAtomicCounter>(
+            _spill_read_bytes_metric.get()));
+
+    INT_UGAUGE_METRIC_REGISTER(_entity, spill_io_thread_pool_max_threads);
+    INT_UGAUGE_METRIC_REGISTER(_entity, spill_io_thread_pool_active_threads);
+    INT_UGAUGE_METRIC_REGISTER(_entity, spill_io_thread_pool_pool_max_queue_size);
+    INT_UGAUGE_METRIC_REGISTER(_entity, spill_io_thread_pool_queue_size);
+
+    _entity->register_hook("spill_io_thread_pool_max_threads", [&]() {
```
Comment (yiguolei) [inline be/src/vec/spill/spill_stream_manager.cpp:120]: thread pool的别加了，zhiqiang 那里给所有的thread pool 统一加

File: be/src/vec/spill/spill_reader.cpp [hunk 3]
```diff
@@ -87,7 +89,9 @@ Status SpillReader::open() {
 
     RETURN_IF_ERROR(file_reader_->read_at(read_offset, result, &bytes_read));
     DCHECK(bytes_read == block_count_ * sizeof(size_t));
-    COUNTER_UPDATE(_read_file_size, bytes_read);
+    total_read_bytes += bytes_read;
+    COUNTER_UPDATE(_read_file_size, total_read_bytes);
+    ExecEnv::GetInstance()->spill_stream_mgr()->updat_spill_read_bytes(total_read_bytes);
```
Comment (yiguolei) [inline be/src/vec/spill/spill_reader.cpp:94]: typo error

**Pull request merged**